src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green'| 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Some of the lengths are lover or equal 0');
+    }
+
+    if (this.a >= b + c || this.b >= a + c || this.c >= a + b) {
+      throw new Error('Side should be less then sum of each other sides');
+    }
+  }
+
+  getArea(): number {
+    const semiPerimetr = 0.5 * (this.a + this.b + this.c);
 
+    const area = Math.sqrt(
+      semiPerimetr
+      * (semiPerimetr - this.a)
+      * (semiPerimetr - this.b)
+      * (semiPerimetr - this.c),
+    );
+
+    return +area.toFixed(2);
```

Comment:
+area.toFixed(2)

when you want to change your type to a number, better to use object Number(); It is better for reading.

Reply:
👍